blockchain/version.go

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) 2013-2016 The btcsuite developers
+// Copyright (c) 2015-2016 The Decred developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+package blockchain
+
+import "fmt"
+
+// calcNextStakeVersion calculates the next stake version based on the provided
+// node.  The next best stake version is defined as a 95% super majority of all
+// stake versions at a StakeDiffWindowSize interval.
+//
+// If the node is not at a StakeDiffWindowSize it returns the prior version.
+//
+// This function is safe for concurrent access.
+func (b *BlockChain) calcNextStakeVersion(node *blockNode) (uint32, error) {
+	if node == nil {
+		return 0, AssertError(fmt.Sprintf(
```

Comment:
No need for `fmt.Sprintf` since it's just a literal.

```Go
return 0, AssertError("calcNextStakeVersion: invalid node")
```

Although that error will not be useful if it's ever hit since it contains no details at all.  I'm of the mind that it shouldn't be checked at all here since it will just panic below anyways when it tries to access `node.height`.  That is the model used throughout the rest of the package.